lib/date_time/date_time.c

```diff
@@ -153,19 +155,29 @@ static int sntp_time_request(struct ntp_servers *server, uint32_t timeout,
 	hints.ai_socktype = SOCK_DGRAM;
 	hints.ai_protocol = 0;
 
-	if (server->addr == NULL) {
+	if (server->addrlen == 0) {
```

Comment:
Could we use the `server->addr.sa_family` field to verify if the address is already stored instead of introducing an extra variable? `AF_UNSPEC` (`0`) could indicate that the structure is empty.

Reply:
> instead of introducing an extra variable

The variable is needed anyway in line 180. I've just used it here, because it was easier than looking inside `server->addr`. I can change this condition if you prefer, but I cannot remove `addrlen` field from the structure.

Reply:
Right, I missed that the struct size is required in `sntp_init`. One could wonder if we could re-use the `sa_family` again to determine the struct size, but I guess it's just a personal preference so no push.

Reply:
We could do that, but it would require some error conditions handling when size returned by `getaddrinfo` differs from expected size for given `sa_family` value. I prefer to keep it simple with one additional variable. We can optimize for memory usage, if necessary.